### PR TITLE
Allow to specify a docheader file location

### PR DESCRIPTION
**Pull request by malukenho**
Linking #15

### CODE REVIEW

File: src/Command/Checker.php [hunk 1]
```diff
@@ -94,8 +96,23 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
-        return $headerValidator->__invoke($fileContent) || false !== strpos($fileContent, $this->header);
+        return $headerValidator->__invoke($fileContent) || false !== strpos($fileContent, $docheaderFile);
+    }
+
+    private function getDocheaderFileContent(InputInterface $input)
+    {
+        $docheaderFile = $input->getOption('docheader');
+
+        if ('.docheader' === $docheaderFile) {
+            $docheaderFile = getcwd() . '/' . $docheaderFile;
```
Comment (weierophinney) [inline src/Command/Checker.php]: You may want to use `DIRECTORY_SEPARATOR` instead to ensure this works on Windows and other non-POSIX systems.

File: src/Command/Checker.php [hunk 2]
```diff
@@ -94,8 +96,23 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
     {
-        return $headerValidator->__invoke($fileContent) || false !== strpos($fileContent, $this->header);
+        return $headerValidator->__invoke($fileContent) || false !== strpos($fileContent, $docheaderFile);
+    }
+
+    private function getDocheaderFileContent(InputInterface $input)
+    {
+        $docheaderFile = $input->getOption('docheader');
+
+        if ('.docheader' === $docheaderFile) {
```
Comment (weierophinney) [inline src/Command/Checker.php]: What happens if a file is specified with no path information? Shouldn't these be treated the same as `.docheader`, and prepend the `PWD`?

File: src/Command/Checker.php [hunk 3]
```diff
@@ -94,8 +96,23 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
+            $docheaderFile = getcwd() . '/' . $docheaderFile;
+        }
+
+        $docheader = (new DocheaderFileResolution())->__invoke($docheaderFile);
```
Comment (weierophinney) [inline src/Command/Checker.php]: This could be more simply written as `(new DocheaderFileResolution())($docheaderFile)`.

Comment (weierophinney) [reply src/Command/Checker.php]: Alternately, have the class redefine its `__invoke()` method as something like `resolve()`.

**Review by weierophinney — CHANGES_REQUESTED** [review]

**Comment by malukenho** [comment]
@weierophinney back to you

**Review by weierophinney — APPROVED** [review]

**Pull request merged**